### PR TITLE
[kube-prometheus-stack] Enable network policy for prometheus

### PR DESCRIPTION
**Pull request by QuentinBisson**
#### What this PR does / why we need it

This PR adds a Network Policy to prometheus in the kube-prometheus-stack so it can scrape components behind a deny all network policy

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

File: charts/kube-prometheus-stack/Chart.yaml
Comment (monotek) [inline charts/kube-prometheus-stack/Chart.yaml:24]:
```suggestion
version: 44.5.0
```

File: charts/kube-prometheus-stack/templates/prometheus/networkpolicy.yaml
Comment (monotek) [inline charts/kube-prometheus-stack/templates/prometheus/networkpolicy.yaml:14]: default can be defined in the values yaml.
i would likely not use "allow all" by default but give a set of reasonable policies.

**Review by monotek — CHANGES_REQUESTED** [review]

File: charts/kube-prometheus-stack/templates/prometheus/networkpolicy.yaml
Comment (QuentinBisson) [reply charts/kube-prometheus-stack/templates/prometheus/networkpolicy.yaml:14]: Pushed a new version with limited default ingress

Comment (monotek) [reply charts/kube-prometheus-stack/templates/prometheus/networkpolicy.yaml:14]: We might want to make the whole .spec part configuralbe by using "{{ toYaml ...}}"?
Just in case somebody likes to change selectors too?

egress default for the operator would be kube api. 
prometheus should be able to reach 9090 for the scrapes. 
Not sure if i miss something?

We could also use deny all as default so everybody has to care individually if networks policy is enabled.

Comment (QuentinBisson) [reply charts/kube-prometheus-stack/templates/prometheus/networkpolicy.yaml:14]: I would think that someone wanting to enable the netpol would have to change the egress anyway unless we set it to allow all. As enablement is a conscious choice, let's limits it to 9090, the kube api and alertmanager then. 
I'll try to figure out how :D 

Regarding the selectors, maybe we allow them to override them for now but not the whole spec? Otherwise we would need to do the same change for prometheus-operator network policies :)

Comment (monotek) [reply charts/kube-prometheus-stack/templates/prometheus/networkpolicy.yaml:14]: Just wanted to go the easy path instead defining overrides for every available option.
In the end i don't much of an opinion here.
Maybe wait for some feedback of the charts maintainers.

Comment (QuentinBisson) [reply charts/kube-prometheus-stack/templates/prometheus/networkpolicy.yaml:14]: @scottrigby @Xtigyro does any one of you have ideas?

Otherwise let's make it deny  or allow all and users will override it to fit their needs

Comment (QuentinBisson) [reply charts/kube-prometheus-stack/templates/prometheus/networkpolicy.yaml]: @monotek I made the egress deny all by default, letting the users override it for now. This could be revisited later

**Review by monotek — APPROVED** [review]

**Comment by QuentinBisson** [comment]
Bumping chart version @monotek can you review? I'm still not approved to merge this repo :(

**Comment by QuentinBisson** [comment]
@monotek I think you forgot to sign your commit :D

**Comment by monotek** [comment]
yes, damn web editor :/
can you please rebase / squash my commit?

**Comment by QuentinBisson** [comment]
Sure :)

**Comment by QuentinBisson** [comment]
@monotek should be good now

**Review by monotek — APPROVED** [review]

**Pull request merged**
